lighthouse-core/audits/page-execution-timings.js

```diff
@@ -0,0 +1,171 @@
+  'Cancel Idle Callback': group.scripting,
+  'Compile Script': group.scriptParseCompile,
+  'Composite Layers': group.compositing,
+  'Console Time': group.scripting,
+  'Image Decode': group.images,
+  'Draw Frame': group.painting,
+  'Embedder Callback': group.scripting,
+  'Evaluate Script': group.scripting,
+  'Event': group.scripting,
+  'Animation Frame Fired': group.scripting,
+  'Fire Idle Callback': group.scripting,
+  'Function Call': group.scripting,
+  'DOM GC': group.scriptGC,
+  'GC Event': group.scriptGC,
+  'GPU': group.gpu,
+  'Hit Test': group.compositing,
+  'Invalidate Layout': group.styleLayout,
+  'JS Frame': group.scripting,
+  'Input Latency': group.scripting,
+  'Layout': group.styleLayout,
+  'Major GC': group.scriptGC,
+  'DOMContentLoaded event': group.scripting,
+  'First paint': group.painting,
+  'FMP': group.painting,
+  'FMP candidate': group.painting,
+  'Load event': group.scripting,
+  'Minor GC': group.scriptGC,
+  'Paint': group.painting,
+  'Paint Image': group.images,
+  'Paint Setup': group.painting,
+  'Parse Stylesheet': group.parseHTML,
+  'Parse HTML': group.parseHTML,
+  'Parse Script': group.scriptParseCompile,
+  'Other': group.other,
+  'Rasterize Paint': group.painting,
+  'Recalculate Style': group.styleLayout,
+  'Request Animation Frame': group.scripting,
+  'Request Idle Callback': group.scripting,
+  'Request Main Thread Frame': group.painting,
+  'Image Resize': group.images,
+  'Finish Loading': group.loading,
+  'Receive Data': group.loading,
+  'Receive Response': group.loading,
+  'Send Request': group.loading,
+  'Run Microtasks': group.scripting,
+  'Schedule Style Recalculation': group.styleLayout,
+  'Scroll': group.compositing,
+  'Task': group.other,
+  'Timer Fired': group.scripting,
+  'Install Timer': group.scripting,
+  'Remove Timer': group.scripting,
+  'Timestamp': group.scripting,
+  'Update Layer': group.compositing,
+  'Update Layer Tree': group.compositing,
+  'User Timing': group.scripting,
+  'Create WebSocket': group.scripting,
+  'Destroy WebSocket': group.scripting,
+  'Receive WebSocket Handshake': group.scripting,
+  'Send WebSocket Handshake': group.scripting,
+  'XHR Load': group.scripting,
+  'XHR Ready State Change': group.scripting,
+};
+
+class PageExecutionTimings extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'page-execution-timings',
+      description: 'Page execution time is high',
+      informative: true,
+      helpText: 'Consider reducing the time spent parsing, compiling and executing JS.' +
+        'You may find delivering smaller JS payloads helps with this.',
+      requiredArtifacts: ['traces'],
+    };
+  }
+
+  /**
+   * @param {!Array<TraceEvent>=} trace
+   * @return {!Map<string, Number>}
+   */
+  static getExecutionTimingsByCategory(trace) {
+    const timelineModel = new DevtoolsTimelineModel(trace);
```

Comment:
is it possible to limit to events before TTCI? Not sure how to do this with `DevtoolsTimelineModel`

Reply:
discussed offline. this could be done in a followup.. it wouldn't make a huge difference though it will be nice.